### PR TITLE
PE-763: Adds support for uniswap v3 to the auction-demo-keeper

### PR DESCRIPTION
**Pull request by godsflaw**
Adds uniswap v3 support.

### CODE REVIEW

**Review by talbaneth — COMMENTED** [review]
Looks good, see small cleanups/formatting related comments.

File: config/mainnet.json [hunk 1]
```diff
@@ -91,24 +114,43 @@
...
+      "name": "RENBTC-A",
+      "decimals": 8,
+      "erc20addr": "0xEB4C2781e4ebA804CE9a9803C67d0893436bB27D",
+      "joinAdapter": "0xFD5608515A47C37afbA68960c1916b79af9491D0",
+      "clipper": "0x834719BEa8da68c46484E001143bDDe29370a6A3",
+      "uniswapV3Callee": "0xdB9C76109d102d2A1E645dCa3a7E671EBfd8e11A",
+      "uniV3Path": [
+        {
+          "fee": 500,
+          "tokenA": "0xEB4C2781e4ebA804CE9a9803C67d0893436bB27D",
```
[the remark this answers is not in the document]
Comment (godsflaw) [reply config/mainnet.json:141]: I used their site, which uses the dynamic router to give the optimal path.  Perhaps this path isn't the deepest liquidity, but at the time, it was the optimal route.

File: config/mainnet.json [hunk 2]
```diff
@@ -91,24 +114,43 @@
...
-      "decimals": 18,
-      "erc20addr": "0x514910771AF9Ca656af840dff83E8264EcF986CA",
-      "joinAdapter": "0xdFccAf8fDbD2F4805C174f856a317765B49E4a50",
-      "clipper": "0x832Dd5f17B30078a5E46Fdb8130A68cBc4a74dC0",
-      "uniswapCallee": "0x49399BB0Fcb52b32aB5A0909206BFf7B54FF80b3",
-      "uniswapRoute": [
-        "0x514910771AF9Ca656af840dff83E8264EcF986CA",
-        "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
-        "0x6B175474E89094C44Da98b954EedeAC495271d0F"
+    "RENBTC-A": {
```
[the remark this answers is not in the document]
Comment (godsflaw) [reply config/mainnet.json:131]: git diff being annoying.

File: config/mainnet-uniswapv2.json [hunk 3]
```diff
@@ -0,0 +1,309 @@
+{
```
[the remark this answers is not in the document]
Comment (godsflaw) [reply config/mainnet-uniswapv2.json:1]: I think they are more an active example, but I could make an examples directory.

File: config/mainnet.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (godsflaw) [reply config/mainnet.json:141]: Going to leave this one for now.

Screen Shot 2021-12-10 at 10 27 18 AM

File: config/mainnet.json [hunk 4]
```diff
@@ -117,37 +159,58 @@
...
-        "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
-        "0x6B175474E89094C44Da98b954EedeAC495271d0F"
+    "YFI-A": {
+      "name": "YFI-A",
+      "decimals": 18,
+      "erc20addr": "0x0bc529c00C6401aEF6D220BE8C6Ea1667F6Ad93e",
+      "joinAdapter": "0x3ff33d9162aD47660083D7DC4bC02Fb231c81677",
+      "clipper": "0x9daCc11dcD0aa13386D295eAeeBBd38130897E6f",
+      "uniswapV3Callee": "0xdB9C76109d102d2A1E645dCa3a7E671EBfd8e11A",
+      "uniV3Path": [
```
[the remark this answers is not in the document]
Comment (godsflaw) [reply config/mainnet.json:203]: You're right, this should be V2 liquidity.  This was a bug, good catch!  I missed that the liquidity was from V2 and not V3 when putting this route in.

[the remark this answers is not in the document]
Comment (godsflaw) [reply config/mainnet.json:203]: Screen Shot 2021-12-10 at 10 31 47 AM

File: package.json [hunk 5]
```diff
@@ -25,6 +25,8 @@
     "@ethersproject/experimental": "5.0.7",
     "@makerdao/dai": "^0.32.11",
     "@makerdao/dai-plugin-mcd": "^1.6.21",
+    "@uniswap/v3-periphery": "^1.3.0",
+    "@uniswap/v3-sdk": "^3.6.3",
```
[the remark this answers is not in the document]
Comment (godsflaw) [reply package.json:29]: You can see all the things I tried and removed.

**Review by talbaneth — APPROVED** [review]
LGTM

**Pull request merged**